lib/cylc/suite_srv_files_mgr.py

```diff
@@ -468,20 +468,19 @@ def register(self, reg=None, source=None, redirect=False):
                     "ERROR: the name '%s' already points to %s.\nUse "
                     "--redirect to re-use an existing name and run "
                     "directory." % (reg, orig_source))
-            sys.stderr.write(
-                "WARNING: the name '%(reg)s' points to %(old)s.\nIt will now"
+            LOG.warning(
+                "the name '%(reg)s' points to %(old)s.\nIt will now"
                 " be redirected to %(new)s.\nFiles in the existing %(reg)s run"
-                " directory will be overwritten.\n" % {
-                    'reg': reg, 'old': orig_source, 'new': source})
+                " directory will be overwritten.\n",
+                {'reg': reg, 'old': orig_source, 'new': source})
             # Remove symlink to the original suite.
             os.unlink(target)
 
         # Create symlink to the suite, if it doesn't already exist.
         if source != orig_source:
             os.symlink(source, target)
 
-        print 'REGISTERED %s -> %s' % (reg, source)
-        return reg
+        return reg, source
```

Comment:
This increases the purity of the method:
* Let it return all the useful variables to the caller.
* Let the caller decide if it needs to print any diagnostics.
* I would have wanted to remove the remaining logging logic if there is an easy way to do so.

Reply:
My first impression was a bit of confusion. Why should we return `source`, being it is an `argument` for the function in the first place (which can be a code smell, or a feature or per design [e.g. builder pattern]).

Then I realized `reg` is also a parameter. I think the code would be simpler to read if it did not return anything. Simply received parameters, registered suites and threw exceptions if any error occurred.

The caller of the function would be responsible for choosing the right value for `reg` (i.e. if you don't have one, use `os.path.basename(os.getcwd())`), and for `source`.

This way, IMO, it would be easier to i) read and understand what the function does, ii) write tests, iii) maintain the code.

The _iii_ may not be clear right now, but if we decide to adopt other strategies for choosing the possible default value for `reg` or `source` (e.g. say we have AWS and decide to use an environment variable for the default values?) we wouldn't have to touch this function, but move the logic to the caller.

So I agree with the change, for the current design of the function :grimacing: but later will try to come up with alternatives (unless I'm missing something that prevents other designs).